packages/framework-provider-azure/src/library/events-adapter.ts

```diff
@@ -180,3 +180,31 @@ export async function storeSnapshot(
   logger.debug('Snapshot stored', snapshotEnvelope)
   return persistableEntitySnapshot
 }
+
+export async function storeDispatchedEvent(
+  cosmosDb: CosmosClient,
+  eventEnvelope: EventEnvelope,
+  config: BoosterConfig
+): Promise<boolean> {
+  const logger = getLogger(config, 'events-adapter#storeDispatchedEvent')
+  logger.debug('[EventsAdapter#storeDispatchedEvent] Storing EventEnvelope for event with ID: ', eventEnvelope.id)
+  try {
+    await cosmosDb
+      .database(config.resourceNames.applicationStack)
+      .container(config.resourceNames.dispatchedEventsStore)
+      .items.create({
+        eventId: eventEnvelope.id,
+      })
+    return true
+  } catch (e) {
+    if (e.code === 409) {
+      // If an item with the same ID already exists in the container, it will return a 409 status code.
+      // See https://learn.microsoft.com/en-us/rest/api/cosmos-db/http-status-codes-for-cosmosdb
+      logger.debug('[EventsAdapter#storeDispatchedEvent] Event has already been dispatched', eventEnvelope.id)
+      return false
+    } else {
+      logger.error('[EventsAdapter#storeDispatchedEvent] Error storing dispatched event', e)
+      return true
```

Comment:
Here we should let the error bubble up so that the core can act upon it if needed (or log it), as other methods do. Basically rethrow it:
```suggestion
      throw e
```

Reply:
Changed in commit 1eb76723145dabcaf935257523f9743dc66f23e7